### PR TITLE
[MINOR] Simplify ProcBuilder code

### PR DESCRIPTION
**Pull request by MyLanPangzi**
MyLanPangzi [Closes #412](https://github.com/yaooqinn/kyuubi/pull/412)    Target Issue Test Plan [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

remove redundancy code, beause scala SAM will opt the code.

### _Why are the changes needed?_
just simpfy code

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
Thanks for your first contribution and welcome! Let's make the CI happy first.

**Comment by yaooqinn** [comment]
```
[ERROR] /home/travis/build/yaooqinn/kyuubi/kyuubi-main/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:108: object creation impossible, since method run in trait Runnable of type ()Unit is not defined
[ERROR] one error found
```

FYI, please check the CI errors @MyLanPangzi 
also cc @pan3793 since I'm not good at scala SAM

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Hi @MyLanPangzi which email are you binding to the github account? 
- 1251723871@qq.com
- hiscat <46845236+MyLanPangzi@users.noreply.github.com>

**Review by pan3793 — APPROVED** [review]

File: kyuubi-main/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala [hunk 1]
```diff
@@ -105,8 +105,8 @@ trait ProcBuilder {
     val proc = processBuilder.start()
     val reader = Files.newBufferedReader(engineLog.toPath, StandardCharsets.UTF_8)
 
-    val redirect = new Runnable {
-      override def run(): Unit = try {
+    val redirect = new Runnable { () =>
```
[the remark this answers is not in the document]
Comment (pan3793) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala]: `_ =>` should match something like:
```
@FunctionalInterface
public interface Consumer<T> {
    void accept(T t);
}
```

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0

**Comment by yaooqinn** [comment]
> this [1251723871@qq.com](mailto:1251723871@qq.com).

You'd better configure your local git `user.email` with it, it's good for you.
